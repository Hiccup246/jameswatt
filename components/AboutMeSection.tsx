import EmailIcon from "./EmailIcon";
import GithubIcon from "./GithubIcon";
import LinkedInIcon from "./LinkedInIcon";
import ResumeIcon from "./ResumeIcon";

export default function AboutMeSection() {
    return (
        <div className="w-full bg-white flex flex-row justify-center pt-10 pb-28">
            <div className="flex flex-col justify-start w-9/12 mx-auto">
                <h1 className="self-center text-3xl font-bold mb-10">Who am I?</h1>
                
                <p className="self-center text-lg text-center font-serif mb-8">
                    I'm a software engineer based out of London, England and an xpat of New Zealand. I enjoy
                    software engineering for its sometimes mind bending mix of subjectiveness and objectivness
                    where there is almost never a sliver bullet. This has helped me develop extensive software
                    skill over 6 years of programming and 3 years of working in high performing agile teams.
                </p>

                <p className="self-center text-lg text-center font-serif">
                    Outside of software my interests include writing, journalism, hiking throughout the world,
                    exploring Londons historic pubs, running, reading books relating to topics such as sociology,
                    fantasy, politics, philosophy and...sensory deprivation tanks ;)
                </p>

                <div className="flex flex-row justify-center mt-10">
                    <div className="w-8 my-auto mr-8 hover:scale-125 transition duration-150">
                        <GithubIcon src="https://www.google.com" />
                    </div>

                    <div className="w-8 my-auto mr-8 hover:cursor-pointer hover:scale-125 duration-200 ease-in">
                        <LinkedInIcon src="https://www.google.com" />
                    </div>

                    <div className="w-10 my-auto mr-8 hover:cursor-pointer hover:scale-125 duration-200 ease-in">
                        <EmailIcon src="https://www.google.com" />
                    </div>

                    <div className="w-10 my-auto hover:cursor-pointer hover:scale-125 duration-200 ease-in">
                        <ResumeIcon src="https://www.google.com" />
                    </div>
                </div>
            </div>
        </div>
    );
}