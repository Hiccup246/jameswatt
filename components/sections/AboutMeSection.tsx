import EmailIcon from "../Icons/EmailIcon";
import GithubIcon from "../Icons/GithubIcon";
import LinkedInIcon from "../Icons/LinkedInIcon";
import ResumeIcon from "../Icons/ResumeIcon";
import SectionLayout from "../layouts/SectionLayout";

export default function AboutMeSection() {
  function copyToClipboard(text: string): void {
    navigator.clipboard.writeText(text);
  }

  return (
    <SectionLayout>
      <h1 className="mb-16 text-center text-2xl font-bold sm:text-3xl">
        Who am I?
      </h1>

      <p className="mb-8 self-center text-center font-serif text-lg">
        I&apos;m a software engineer based out of London, England and an xpat of
        New Zealand. I enjoy software engineering for its sometimes mind bending
        mix of subjectiveness and objectivness where there is almost never a
        sliver bullet. This has helped me develop extensive software skill over
        6 years of programming and 3 years of working in high performing agile
        teams.
      </p>

      <p className="self-center text-center font-serif text-lg">
        Outside of software my interests include writing, journalism, hiking
        throughout the world, exploring Londons historic pubs, running, reading
        books relating to topics such as sociology, fantasy, politics,
        philosophy and...sensory deprivation tanks ;)
      </p>

      <div className="mt-16 flex flex-row justify-center">
        <div className="my-auto mr-8 w-8 duration-150 ease-in hover:-translate-y-0.5">
          <a
            className="duration-150 ease-in hover:text-grey"
            href="https://github.com/Hiccup246"
            aria-label="Github Profile"
            title="Github Profile"
          >
            <GithubIcon />
          </a>
        </div>

        <div className="my-auto mr-8 w-8 duration-150 ease-in hover:-translate-y-0.5">
          <a
            className="duration-150 ease-in hover:text-grey"
            href="https://www.linkedin.com/in/jameswattpro"
            aria-label="LinkedIn Profile"
            title="LinkedIn Profile"
          >
            <LinkedInIcon />
          </a>
        </div>

        <div className="group relative my-auto mr-8 w-10">
          <div className="duration-200 ease-in group-hover:-translate-y-0.5 group-hover:text-grey">
            <EmailIcon />
          </div>

          <div
            role="tooltip"
            className="absolute top-[50px] -left-[128px] h-fit w-[246px] rounded-lg bg-brown opacity-0 shadow delay-200 group-hover:opacity-100"
          >
            <div className="relative">
              <div className="absolute right-[90px] -top-[22px] bottom-2 z-10 h-4 w-4 rotate-45 bg-brown" />

              <div className="z-20 my-3 flex w-full justify-center">
                <a
                  className="mr-2 p-1 hover:underline"
                  href="mailto:james@jameswatt.io"
                  target="_blank"
                  rel="noopener noreferrer"
                  aria-label="Email Address"
                  title="James Watt Email"
                >
                  Open in email
                </a>

                <button
                  onClick={() => copyToClipboard("james@jameswatt.io")}
                  className="z-20 ml-2 h-fit w-fit rounded bg-darkbrown p-1 px-2 hover:opacity-70 active:scale-105"
                >
                  Copy email
                </button>
              </div>
            </div>
          </div>
        </div>

        <div className="my-auto w-10 duration-150 ease-in hover:-translate-y-0.5">
          <a
            className="duration-150 ease-in hover:text-grey"
            href="/jameswatt-resume.pdf"
            target="_blank"
            rel="noreferrer"
            aria-label="PDF Resume"
            title="James Watt PDF C.V."
          >
            <ResumeIcon />
          </a>
        </div>
      </div>
    </SectionLayout>
  );
}
