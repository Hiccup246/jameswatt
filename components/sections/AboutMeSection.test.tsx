import AboutMeSection from "./AboutMeSection";
import { render, screen } from "@testing-library/react";

describe("AboutMeSection", () => {
  const env = process.env;

  afterEach(() => {
    process.env = { ...env };
  });

  it("renders correctly", () => {
    const { container } = render(<AboutMeSection />);
    expect(container).toMatchSnapshot();
  });

  it("Copy email button copies AUTHOR_CONTACT_EMAIL env variable to clipboard", () => {
    process.env.NEXT_PUBLIC_AUTHOR_CONTACT_EMAIL = "john@aol.com";
    const mockWriteText = jest.fn((_text) => {});
    Object.defineProperty(navigator, "clipboard", {
      value: { writeText: mockWriteText },
    });
    render(<AboutMeSection />);

    screen.getByRole("button").click();

    expect(mockWriteText).toHaveBeenCalledWith("john@aol.com");
  });
});
