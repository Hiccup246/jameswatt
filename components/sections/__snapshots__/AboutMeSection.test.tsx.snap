// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutMeSection renders correctly 1`] = `
<div>
  <div
    class="flex w-full flex-row justify-center pt-12 pb-20 pt-20 bg-white dark:bg-darkteal dark:text-softwhite"
  >
    <section
      class="mx-auto flex w-9/12 flex-col justify-start"
    >
      <h1
        class="mb-16 text-center text-2xl font-bold sm:text-3xl"
      >
        Who am I?
      </h1>
      <p
        class="mb-8 self-center text-center font-serif text-lg"
      >
        I'm a software engineer based out of London, England and an xpat of New Zealand. I enjoy software engineering for its sometimes mind bending mix of subjectiveness and objectivness where there is almost never a sliver bullet. This has helped me develop extensive software skill over 6 years of programming and 3 years of working in high performing agile teams.
      </p>
      <p
        class="self-center text-center font-serif text-lg"
      >
        Outside of software my interests include writing, journalism, hiking throughout the world, exploring Londons historic pubs, running, reading books relating to topics such as sociology, fantasy, politics, philosophy and...sensory deprivation tanks ;)
      </p>
      <div
        class="mt-16 flex flex-row justify-center"
      >
        <div
          class="group my-auto mr-8 w-8"
        >
          <a
            aria-label="Github Profile"
            class="block duration-150 ease-in group-hover:-translate-y-0.5 group-hover:text-grey group-hover:text-grey"
            href="https://github.com/Hiccup246"
            title="Github Profile"
          >
            <svg
              viewBox="0 0 496 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                fill="currentColor"
              />
            </svg>
          </a>
        </div>
        <div
          class="group my-auto mr-8 w-8"
        >
          <a
            aria-label="LinkedIn Profile"
            class="block duration-150 ease-in group-hover:-translate-y-0.5 group-hover:text-grey group-hover:text-grey"
            href="https://www.linkedin.com/in/jameswattpro"
            title="LinkedIn Profile"
          >
            <svg
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
                fill="currentColor"
              />
            </svg>
          </a>
        </div>
        <div
          class="group relative my-auto mr-8 w-10"
          tabindex="0"
        >
          <div
            class="duration-200 ease-in group-hover:-translate-y-0.5 group-hover:text-grey"
          >
            <svg
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48H48zM0 176V384c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V176L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="invisible absolute -left-[128px] top-[50px] h-fit w-[246px] rounded-lg bg-brown opacity-0 shadow delay-200 duration-200 group-focus-within:visible group-focus-within:opacity-100 group-hover:visible group-hover:opacity-100 dark:bg-darkgrey"
            role="tooltip"
          >
            <div
              class="relative"
            >
              <div
                class="absolute -top-[20px] right-[90px] z-10 h-4 w-4 rotate-45 bg-brown dark:bg-darkgrey"
              />
              <div
                class="z-20 my-3 flex w-full justify-center"
              >
                <a
                  aria-label="Email Address"
                  class="mr-2 p-1 hover:underline"
                  href="mailto:undefined"
                  rel="noopener noreferrer"
                  target="_blank"
                  title="James Watt Email"
                >
                  Open in email
                </a>
                <button
                  class="z-20 ml-2 h-fit w-fit rounded bg-darkbrown p-1 px-2 hover:opacity-70 active:-translate-y-0.5 dark:bg-darkteal"
                >
                  Copy email
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="group my-auto w-10"
        >
          <a
            aria-label="PDF Resume"
            class="block duration-150 ease-in group-hover:-translate-y-0.5 group-hover:text-grey"
            href="/jameswatt-resume.pdf"
            rel="noreferrer"
            target="_blank"
            title="James Watt PDF C.V."
          >
            <svg
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M368 344h96c13.25 0 24-10.75 24-24s-10.75-24-24-24h-96c-13.25 0-24 10.75-24 24S354.8 344 368 344zM208 320c35.35 0 64-28.65 64-64c0-35.35-28.65-64-64-64s-64 28.65-64 64C144 291.3 172.7 320 208 320zM512 32H64C28.65 32 0 60.65 0 96v320c0 35.35 28.65 64 64 64h448c35.35 0 64-28.65 64-64V96C576 60.65 547.3 32 512 32zM528 416c0 8.822-7.178 16-16 16h-192c0-44.18-35.82-80-80-80h-64C131.8 352 96 387.8 96 432H64c-8.822 0-16-7.178-16-16V160h480V416zM368 264h96c13.25 0 24-10.75 24-24s-10.75-24-24-24h-96c-13.25 0-24 10.75-24 24S354.8 264 368 264z"
                fill="currentColor"
              />
            </svg>
          </a>
        </div>
      </div>
    </section>
  </div>
</div>
`;
